### PR TITLE
[POKEMON-0000] Created Pokemon Model Class

### PR DESCRIPTION
**Pull request by arriyam**
## Details

Create pure model `Pokemon` class.

## Related Tickets

Ticket: [POKEMON-0000](https://trello.com/c/nHo60qFd/2-implement-pokemon-data-retrieval-and-model)
Epic ticket: [POKEMON](https://trello.com/b/nsSxCgvo/pokemon)

## Motivation and Context

Need a pure Pokemon model class to structure the Pokemon data. This is needed to for reusability and ease of working with multiple Pokemon.

### CODE REVIEW

File: PokemonRampUp/Model/Pokemon.swift [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+//  Pokemon.swift
+//  PokemonRampUp
+//
+//  Created by Thushen Mohanarajah on 2023-02-10.
+//
+
+import Foundation
+import UIKit
+
+class Pokemon {
```
Comment (blavkjay) [inline PokemonRampUp/Model/Pokemon.swift]: you should use a struct for your models.

read more here: https://stackoverflow.com/questions/24232799/why-choose-struct-over-class

**Review by therealchriswong — COMMENTED** [review]
Why a class?

File: PokemonRampUp/Model/Pokemon.swift [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    let name: String
+    let type: String
+    let hiddenAbilites: [String]
+    let image: UIImage
+    let moves: [String]
+    
+    //
+    // MARK: - Initialization
+    //
+    init(name: String, type: String, hiddenAbilites: [String], image: UIImage, moves: [String]) {
```
Comment (blavkjay) [inline PokemonRampUp/Model/Pokemon.swift]: can you move each property to a new line like this:

init(name: String,
       type: String)

File: PokemonRampUp/Model/Pokemon.swift (hunk 1 above)
Comment (arriyam) [reply PokemonRampUp/Model/Pokemon.swift]: This reading makes sense. From my understanding most languages strive to be pass-by-reference or pass-by-value. Why would Swift openly support both. In Java objects are mostly pass by pass-by-reference while primitive data types are pass-by-value

File: PokemonRampUp/Model/Pokemon.swift [hunk 3]
```diff
@@ -8,7 +8,7 @@
 import Foundation
 import UIKit
 
-class Pokemon {
+struct Pokemon {
     let name: String
     let type: String
     let hiddenAbilites: [String]
```
Comment (blavkjay) [inline PokemonRampUp/Model/Pokemon.swift:14]: is this a typo or it was intentional? hiddenAbilites

**Review by blavkjay — APPROVED** [review]

**Pull request merged**